Source/UserManagement/Read/DataCollectors/DataCollector.cs

```diff
@@ -10,7 +10,7 @@ namespace Read.DataCollectors
     public class DataCollector : IReadModel
     { 
         [BsonId]
```

Comment:
Not part of this commit - but a comment in general. Mapping should not be done in the actual object, but use the Mongo mapping classes:
https://mongodb.github.io/mongo-csharp-driver/2.0/reference/bson/mapping/

Reply:
I see, single responsibility principle in its fullest I guess :wink: I'll look into it

Reply:
👍  Its actually separation of concerns; one being the representation of the readmodel, while the other being a very technology specific mapping and description needed as requirement for Mongo.

You see the similar thing with validation in vanilla ASP.NET MVC where attributes are being used. Validation being a separate concern all together.

When one separates the concerns, it is easier to identity patterns that could then result in the concept of cross-cutting concerns. Patterns are then mapped into conventions and one can have rules in the codebase that things will automatically pick up on instead. Its also possible to apply things cross-cuttingly in a specific manner as well. For instance if you're using a Concept type representing something, you could have an Input validation rule associated with it and it could then be automatically discovered and hooked up whenever it is in use for a Command for instance.